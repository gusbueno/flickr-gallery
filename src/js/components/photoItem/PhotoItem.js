'use strict'

import React, { PureComponent } from 'react'

import styles from './PhotoItem.scss'

class PhotoItem extends PureComponent {
  render () {
    return (
      <div className={styles['photo-item']}>
        <div className={styles['photo-wrapper']}>
          <img src={`https://farm${this.props.photo.farm}.staticflickr.com/${this.props.photo.server}/${this.props.photo.id}_${this.props.photo.secret}.jpg`} alt={this.props.photo.title} />
        </div>
        <div className={styles['photo-info']}>
          <span className={styles['photo-title']}>{this.props.photo.title}</span>
          <div className={styles['photo-owner-wrapper']}>
            <a href={`https://www.flickr.com/photos/${this.props.photo.owner}/${this.props.photo.id}`} target='blank' className={styles['photo-owner-text']}>{this.props.photo.ownername}</a>
          </div>
        </div>
      </div>
    )
  }
}

export default PhotoItem
