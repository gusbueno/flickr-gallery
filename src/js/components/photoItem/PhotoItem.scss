.photo-item {
  width: 24%;
  padding: 10px;
  box-sizing: border-box;
  background-color: #CBDDEF;
  margin-bottom: 10px;
  border-radius: 5px;
  box-shadow: 0px 0px 5px 1px black;

  .photo-wrapper {
    border-radius: 5px;
    background-color: #CBDDEF;
    position: relative;
    max-height: 200px;
    overflow: hidden;
    display: flex;
    align-items: center;
    justify-content: center;
    &:hover {
      cursor: pointer;
    }

    img {
      height: auto;
    }
  }

  .photo-info {
    padding-top: 10px;

    .photo-title {
      padding-top: 10px;
      font-weight: bold;
      font-size: 18px;
      
    }

    .photo-owner-wrapper {
      text-align: right;

      .photo-owner-text {
        font-size: 12px;
      }
    }
  }
}
