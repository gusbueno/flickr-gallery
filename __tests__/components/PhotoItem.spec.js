'use strict'

/* global describe, it, expect, jest */

import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import PhotoItem from '../../src/js/components/photoItem/PhotoItem'

describe('<PhotoItem /> Component', () => {
  const photo = {
    title: 'Title',
    farm: 123,
    server: '123',
    id: '123',
    secret: '123',
    owner: '123',
    ownername: 'Pepe'
  }
  it('should match with the snapshot', () => {
    const tree = renderer.create(
      <PhotoItem photo={photo} onRequestPhotoDetail={() => {}} />
    )
    const json = tree.toJSON()
    expect(json).toMatchSnapshot()
  })

  const onRequestPhotoDetailMock = jest.fn()
  const photoItemShallow = shallow(
    <PhotoItem photo={photo} onRequestPhotoDetail={onRequestPhotoDetailMock} />
  )

  it('should exists', () => {
    expect(photoItemShallow.length).toEqual(1)
  })

  it('should render pagination wrapper', () => {
    expect(photoItemShallow.find('.photo-item').length).toEqual(1)
  })

  it('should call onRequestPhotoDetail with photo object on click photo', () => {
    photoItemShallow.find('.photo-wrapper').first().simulate('click')
    expect(onRequestPhotoDetailMock).toHaveBeenCalledWith(photo)
  })
})
