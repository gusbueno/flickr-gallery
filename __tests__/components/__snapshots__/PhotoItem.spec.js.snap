// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhotoItem /> Component should match with the snapshot 1`] = `
<div
  className="photo-item"
>
  <div
    className="photo-wrapper"
    onClick={[Function]}
  >
    <img
      alt="Title"
      src="https://farm123.staticflickr.com/123/123_123.jpg"
    />
  </div>
  <div
    className="photo-info"
  >
    <span
      className="photo-title"
    >
      Title
    </span>
    <div
      className="photo-owner-wrapper"
    >
      <a
        className="photo-owner-text"
        href="https://www.flickr.com/photos/123/123"
        target="blank"
      >
        Pepe
      </a>
    </div>
  </div>
</div>
`;
